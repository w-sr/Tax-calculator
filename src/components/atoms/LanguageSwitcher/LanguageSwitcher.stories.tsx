import type { Meta, StoryObj } from "@storybook/react";

import { LanguageSwitcher } from "./LanguageSwitcher";

const meta: Meta<typeof LanguageSwitcher> = {
  title: "LanguageSwitcher",
  component: LanguageSwitcher,
};

export default meta;
type Story = StoryObj<typeof LanguageSwitcher>;

export const DefaultSwitcher: Story = {
  render: () => <LanguageSwitcher />,
};
