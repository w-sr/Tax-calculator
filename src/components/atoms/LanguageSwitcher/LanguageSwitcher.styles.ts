import styled from "styled-components";

export const LanguageSwitcher = styled.div``;

export const Select = styled.select``;

export const Option = styled.option``;
