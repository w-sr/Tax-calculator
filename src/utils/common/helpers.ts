const buildPercent = (decimal?: number) =>
  new Intl.NumberFormat("en-US", {
    style: "percent",
    minimumFractionDigits: decimal ?? 2,
  });

/**
 *
 * @param value
 * @param decimal
 * @returns value with percentage
 * @example 0.20 => 20%
 */
export const percentValue = (
  value: number | undefined,
  decimal?: number
): string => {
  if (value === undefined) return "N/A";
  const percent = buildPercent(decimal);
  return percent.format(value);
};

/**
 *
 * @param value
 * @param decimal
 * @returns value with currency
 * @example 20 => $20
 */
export const moneyValue = (
  value: number | undefined,
  decimal?: number
): string => {
  const money = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
    currencyDisplay: "narrowSymbol",
    minimumFractionDigits: decimal ?? 2,
  });
  return money.format(value ?? 0);
};

/**
 *
 * @param min
 * @param max
 * @returns formatted bracket
 * @example min: 0, max: 2000 => "$2000 or less"
 *          min: 2000, max: 4000 => "$2000 to $4000"
 *          min: 4000 => "More than $4000"
 */
export const bracketFormat = (min: number, max?: number): string => {
  if (min === 0) return moneyValue(max, 0) + " or less";
  if (max) return moneyValue(min, 0) + " to " + moneyValue(max, 0);
  else return "More than " + moneyValue(min, 0);
};
