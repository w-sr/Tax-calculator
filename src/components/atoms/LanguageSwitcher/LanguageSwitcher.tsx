import { useTranslation } from "react-i18next";
import * as S from "./LanguageSwitcher.styles";

const LanguageSwitcher = () => {
  const { i18n } = useTranslation();

  return (
    <S.LanguageSwitcher>
      <S.Select
        value={i18n.language}
        onChange={(e) => i18n.changeLanguage(e.target.value)}
      >
        <S.Option value="en">English</S.Option>
      </S.Select>
    </S.LanguageSwitcher>
  );
};

LanguageSwitcher.displayName = "LanguageSwicher";

export { LanguageSwitcher };
