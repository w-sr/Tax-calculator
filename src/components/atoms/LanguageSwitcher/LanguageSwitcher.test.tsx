import { renderWithTheme } from "../../../utils/tests/helpers";
import { LanguageSwitcher } from "./LanguageSwitcher";
import { I18nextProvider } from "react-i18next";
import i18n from "../../../i18n/config";

describe("<LanguageSwitcher />", () => {
  it("should match the snapshot", () => {
    const { container } = renderWithTheme(
      <I18nextProvider i18n={i18n}>
        <LanguageSwitcher />
      </I18nextProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
