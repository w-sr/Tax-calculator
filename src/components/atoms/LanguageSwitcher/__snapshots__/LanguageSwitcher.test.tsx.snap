// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSwitcher /> should match the snapshot 1`] = `
<div
  class="sc-beqWaB"
>
  <select
    class="sc-gueYoa"
  >
    <option
      class="sc-dmqHEX"
      value="en"
    >
      English
    </option>
  </select>
</div>
`;
